pkg/kv/migrations/rbac_to_acl.go

```diff
@@ -216,6 +216,9 @@ func rbacToACL(ctx context.Context, svc auth.Service, doUpdate bool, creationTim
 		if err != nil {
 			return nil, err
 		}
+		if len(users) == 0 { // avoid infinite loop when no users
+			break
+		}
```

Comment:
Note - it should not be infinite loop as the last call should have return empty next token - has more should be false.
It is possible that we get no users and next token - but it is a bug in the API (or not effective) but it should not get to infinite loop.
Saying that not just because of the extra code - I don't think this is the expectation from the API. Not checking the length of the response as a way to break.

Reply:
Fixed